src/DigitalTwin/CodeGeneratorCore.ts

```diff
@@ -580,65 +580,58 @@ export class CodeGeneratorCore {
       md5value = targetConfigItem.codeGeneratorLocation.ubuntuMd5;
     }
 
-    const loading = setInterval(() => {
-      channel.append('.');
-    }, 1000);
-
-    try {
-      // Download
-      utils.channelShowAndAppend(
+    // Download
+    utils.channelShowAndAppend(
+        channel,
+        `Step 1: Downloading ${DigitalTwinConstants.codeGenCli} v${
+            newVersion} package ...`);
+    const downloadOption: request
+        .OptionsWithUri = {method: 'GET', uri: packageUri, encoding: null};
+    const zipData = await request(downloadOption).promise();
+    const tempPath = path.join(os.tmpdir(), FileNames.iotworkbenchTempFolder);
+    const filePath = path.join(tempPath, `${md5value}.zip`);
+    fs.writeFileSync(filePath, zipData);
+    // clearInterval(loading);
+    utils.channelShowAndAppendLine(channel, ' download complete.');
+
+    // Verify
+    // Validate hash code
+    utils.channelShowAndAppend(
+        channel, 'Step 2: Validating hash code for the package ...');
+    const hashvalue = await FileUtility.getFileHash(filePath);
+    if (hashvalue !== md5value) {
+      utils.channelShowAndAppendLine(
           channel,
-          `Step 1: Downloading ${DigitalTwinConstants.codeGenCli} v${
-              newVersion} package ...`);
-      const downloadOption: request
-          .OptionsWithUri = {method: 'GET', uri: packageUri, encoding: null};
-      const zipData = await request(downloadOption).promise();
-      const tempPath = path.join(os.tmpdir(), FileNames.iotworkbenchTempFolder);
-      const filePath = path.join(tempPath, `${md5value}.zip`);
-      fs.writeFileSync(filePath, zipData);
-      clearInterval(loading);
-      utils.channelShowAndAppendLine(channel, ' download complete.');
-
-      // Verify
-      // Validate hash code
-      utils.channelShowAndAppend(
-          channel, 'Step 2: Validating hash code for the package ...');
-      const hashvalue = await FileUtility.getFileHash(filePath);
-      if (hashvalue !== md5value) {
+          `the downloaded ${DigitalTwinConstants.codeGenCli} v${
+              newVersion} package has been corrupted.`);
+      if (installOrUpgrade === 1) {

```

Comment:
What does number 1 means here? Better to use CodeGenCliOperation enum value to better illustrate the meaning.